ui/src/templates/actions/thunks.ts

```diff
@@ -141,37 +89,29 @@ export const createTemplate = (template: DocumentCreate) => async (
 export const createTemplateFromResource = (
   resource: DocumentCreate,
   resourceName: string
-) => async (dispatch, getState: GetState) => {
+) => async (dispatch: Dispatch<Action>, getState: GetState) => {
   try {
     const org = getOrg(getState())
     await client.templates.create({...resource, orgID: org.id})
     dispatch(notify(copy.resourceSavedAsTemplate(resourceName)))
   } catch (e) {
     console.error(e)
-    dispatch(copy.saveResourceAsTemplateFailed(resourceName, e))
+    dispatch(notify(copy.saveResourceAsTemplateFailed(resourceName, e)))
   }
 }
 
-interface SetTemplateSummary {
-  type: ActionTypes.SetTemplateSummary
-  payload: {id: string; templateSummary: TemplateSummary}
-}
-
-export const setTemplateSummary = (
-  id: string,
-  templateSummary: TemplateSummary
-): SetTemplateSummary => ({
-  type: ActionTypes.SetTemplateSummary,
-  payload: {id, templateSummary},
-})
-
 export const updateTemplate = (id: string, props: TemplateSummary) => async (
-  dispatch
+  dispatch: Dispatch<Action>
 ): Promise<void> => {
   try {
```

Comment:
```suggestion
  setTemplateSummary(id, RemoteDataState.Loading)
  try {
```
Probably want to set the loading state here.